tensorflow_probability/python/positive_semidefinite_kernels/polynomial.py

```diff
@@ -0,0 +1,225 @@
+        fitted function goes through the origin. A value of `None` is treated
+        like 0.
+        Default Value: `None`
+      slope_variance: Positive floating point `Tensor` that controls the
+        variance of the regression line slope that is the basis for the
+        polynomial. Must be broadcastable with `bias_variance` and inputs to
+        `apply` and `matrix` methods. A value of `None` is treated like 1.
+        Default Value: `None`
+      exponent: Positive floating point `Tensor` that controls the exponent
+        (also known as the degree) of the polynomial function. Must be
+        broadcastable with `bias_variance`, `slope_variance` and inputs to
+        `apply` and `matrix` methods. A value of `None` is treated like 1
+        which results in a linear kernel.
+        Default Value: `None`
+      feature_ndims: Python `int` number of rightmost dims to include in
+        kernel computation.
+        Default Value: 1
+      validate_args: If `True`, parameters are checked for validity despite
+        possibly degrading runtime performance
+        Default Value: `False`
+      name: Python `str` name prefixed to Ops created by this class.
+        Default Value: `'Polynomial'`
+    """
+    with tf.name_scope(name, values=[bias_variance, slope_variance,
+                                     exponent]) as _:
+      dtype = dtype_util.common_dtype(
+          [bias_variance, slope_variance, exponent], tf.float32)
+      if bias_variance is not None:
+        bias_variance = tf.convert_to_tensor(
+            bias_variance, name='bias_variance', dtype=dtype)
+      self._bias_variance = _validate_arg_if_not_none(
+          bias_variance, tf.assert_positive, validate_args)
+      if slope_variance is not None:
+        slope_variance = tf.convert_to_tensor(
+            slope_variance, name='slope_variance', dtype=dtype)
+      self._slope_variance = _validate_arg_if_not_none(
+          slope_variance, tf.assert_positive, validate_args)
+      if exponent is not None:
+        exponent = tf.convert_to_tensor(
+            exponent, name='exponent', dtype=dtype)
+      self._exponent = _validate_arg_if_not_none(
+          exponent, tf.assert_positive, validate_args)
+      tf.assert_same_float_dtype(
+          [self._bias_variance, self._slope_variance, self._exponent])
+    super(Polynomial, self).__init__(
+        feature_ndims, dtype=dtype, name=name)
+
+  @property
+  def bias_variance(self):
+    """Variance on bias parameter."""
+    return self._bias_variance
+
+  @property
+  def slope_variance(self):
+    """Variance on slope parameter."""
+    return self._slope_variance
+
+  @property
+  def exponent(self):
+    """Exponent of the polynomial term."""
+    return self._exponent
+
+  def _batch_shape(self):
+    scalar_shape = tf.TensorShape([])
+    return tf.broadcast_static_shape(
+        tf.broadcast_static_shape(
+            scalar_shape if self.slope_variance is None
```

Comment:
Consider defining a local helper for this pattern?

def maybe_shape(tensor):
  if tensor is None:
    return scalar_shape
etc.

Here and below.